src/test/java/use_case/execute_buy/ExecuteBuyInteractorTest.java

```diff
@@ -85,10 +114,57 @@ void insufficientBalanceTest() throws ValidationException {
         }
     }
 
+    @Test
+    void stockNotFoundTest() throws ValidationException {
+        User mockUser = createMockUserWithBalance(10000.0);
+
+        try (MockedStatic<MarketTracker> stockMarketMockedStatic = Mockito.mockStatic(MarketTracker.class)) {
+            stockMarketMockedStatic.when(MarketTracker::Instance).thenReturn(marketTrackerMock);
+            when(marketTrackerMock.getStock("AAPL")).thenReturn(Optional.empty());
+
+            ExecuteBuyInputData inputData = new ExecuteBuyInputData("dummy", "AAPL", 10);
+            ExecuteBuyInteractor interactor = new ExecuteBuyInteractor(dataAccess, outputPresenter);
+
+            interactor.execute(inputData);
+
+            // Verify error view was prepared
+            verify(outputPresenter).prepareStockNotFoundExceptionView();
+
+            // Verify no changes were made
+            assertFalse(mockUser.getPortfolio().getUserStock("AAPL").isPresent(),
+                    "Stock should not be in portfolio due to stock not found");
+            assertEquals(10000.0, mockUser.getBalance(), "Balance should remain unchanged");
+            assertTrue(mockUser.getTransactionHistory().getAllTransactions().isEmpty(),
+                    "No transaction should be recorded");
+        }
+    }
+
+//    // TODO: Fail the test
+//    @Test
+//    void validationExceptionTest() throws ValidationException {
```

Comment:
To test `ValidationException`, you can mock a `SessionManager` that always throws a `ValidationException`, and check whether the interactor invokes `prepareValidationExceptionView` and does not modify user data.